frontend/src/main/web/lib/components/glossary/ActionCell.jsx

```diff
@@ -71,13 +101,68 @@ var ActionCell = React.createClass({
 
       var updateButton,
         cancelButton,
-        comment = (
-          <Comment
-            className="ml1/4"
-            readOnly={!this.props.canUpdateEntry || !canUpdateComment || isSaving}
-            value={this.state.entry.transTerm.comment}
-            onUpdateCommentCallback={this._onUpdateComment}/>
+        readOnlyComment = !this.props.canUpdateEntry || !canUpdateComment || isSaving,
+        disableCommentUpdate = !this._hasCommentChanged(),
+        saveCommentButton;
+
+      if(this.state.savingComment) {
+        saveCommentButton = (
+          <Button kind='primary' size={-1} disabled={disableCommentUpdate}
+            onClick={this._onUpdateComment} loading>
+            Update Comment
+          </Button>
+        );
+      } else {
+        saveCommentButton = (
+          <Button kind='primary' size={-1} disabled={disableCommentUpdate}
+            onClick={this._onUpdateComment}>
+            Update Comment
+          </Button>
+        );
+      }
+
+      if(readOnlyComment !== true) {
+        var tooltip = (
+          <Tooltip id="comment" title="Comment">
+            <textarea className="p1/4 w100p bd2 bdcsec30 bdrs1/4"
+              onChange={this._onCommentChange}
+              value={this.state.comment}
+              onKeyUp={this._handleCommentKeyUp}/>
+            <div className="mt1/4">
+              <Button className="mr1/2" link
+                onClick={this._onCancelComment}>
+                Cancel
+              </Button>
+            {saveCommentButton}
+            </div>
+          </Tooltip>
         );
+      } else {
+        var comment = StringUtils.isEmptyOrNull(this.state.comment) ? (<i>No comment</i>) : (<span>{this.state.comment}</span>);
```

Comment:
Looks like `comment` is supposed to be scoped to just this block. Use `let comment =` to do that.